src/main/java/duke/tasks/TaskManager.java

```diff
@@ -0,0 +1,222 @@
+    }
+
+    /**
+     * Sets the todo list with the given argument.
+     * @param todos the todo list to be set.
+     */
+    public void initialiseTodos(List<Todo> todos) {
+        this.todos = todos;
+    }
+
+    /**
+     * Sets the deadline list with the given argument.
+     * @param deadlines the deadline list to be set.
+     */
+    public void initialiseDeadlines(List<Deadline> deadlines) {
+        this.deadlines = deadlines;
+    }
+
+    /**
+     * Sets the event list with the given argument.
+     * @param events the event list to be set.
+     */
+    public void initialiseEvents(List<Event> events) {
+        this.events = events;
+    }
+
+    /**
+     * Adds a todo.
+     * @param todo the todo to be added.
+     * @throws DukeException if something went wrong while saving.
+     */
+    public void addTodo(Todo todo) throws DukeException {
+        todos.add(todo);
+        save();
+    }
+
+    /**
+     * Adds a deadline and sorts the deadline list.
+     * @param deadline the deadline to be added.
+     * @throws DukeException if something went wrong while saving.
+     */
+    public void addDeadline(Deadline deadline) throws DukeException {
+        deadlines.add(deadline);
+        deadlines.sort(Deadline::compareTo);
+        save();
+    }
+
+    /**
+     * Adds a event and sorts the events list.
+     * @param event the event to be added.
+     * @throws DukeException if something went wrong while saving.
+     */
+    public void addEvent(Event event) throws DukeException {
+        events.add(event);
+        events.sort(Event::compareTo);
+        save();
+    }
+
+    /**
+     * Gets each task from the three lists and saves them to Task.txt file.
+     * @throws DukeException if something went wrong while writing to the file.
+     */
+    private void save() throws DukeException {
+        StringBuilder sb = new StringBuilder();
+        for (Todo todo : todos) {
+            sb.append(todo.saveText()).append("\n");
+        }
+        for (Deadline deadline : deadlines) {
+            sb.append(deadline.saveText()).append("\n");
+        }
+        for (Event event : events) {
+            sb.append(event.saveText()).append("\n");
+        }
+        Storage.writeTasksFile(sb.toString());
+    }
+
+    /**
+     * Creates a string representation of all the tasks.
+     * @return the string representation of all the tasks
+     */
+    public String listTasks() {
+        int i = 1;
+        StringBuilder sb = new StringBuilder();
+        if (!todos.isEmpty()) {
+            sb.append("Here are your todos:\n");
+            for (Todo todo : todos) {
+                sb.append(i).append(". ").append(todo).append("\n");
+                i++;
+            }
+            sb.append("\n");
+        } else {
+            sb.append("You have no todos.\n");
+        }
+        if (!deadlines.isEmpty()) {
+            sb.append("Here are your deadlines:\n");
+            for (Deadline deadline : deadlines) {
```

Comment:
This for loop is quite similar with the for loop in line 126, you may consider extracting out as a method? (: